Tools/RunTopTagger.h

```diff
@@ -0,0 +1,235 @@
+        auto* WTLV              = new std::vector<TLorentzVector>();
+        auto* TopJetsMap        = new std::map< int , std::vector<TLorentzVector> >();
+        int nAllTops        = -1;
+        int nMergedTops     = -1;
+        int nSemiMergedTops = -1;
+        int nResolvedTops   = -1;
+        int nWs             = -1;
+
+        //Select AK4 jets to use in tagger
+        //When reading from the resolvedTopCandidate collection from nanoAOD you must pass ALL ak4 jets to ttUtility::ConstAK4Inputs below, 
+        //but we can specify a filter vector to tell it to ignore jets we don't want 
+        std::vector<uint8_t> ak4Filter(Jet_LV.size(), true);
+        for(int i = 0; i < ak4Filter.size(); ++i)
+        {
+            //no need to slow the tagger down considering low pt jets
+            if(Jet_LV[i].Pt() < 20.0) ak4Filter[i] = false;
+
+            //do some logic here to decide which jet was lepton/photon matched
+            bool isLepPhotonMatch = false;
+            if(isLepPhotonMatch) ak4Filter[i] = false;
+        }
+
+        //Select AK8 jets to use in tagger
+        std::vector<uint8_t> ak8Filter(FatJet_LV.size(), true);
+        for(int i = 0; i < ak8Filter.size(); ++i)
+        {
+            //no need to slow the tagger down considering low pt jets
+            if(FatJet_LV[i].Pt() < 200.0) ak8Filter[i] = false;
+
+            //do some logic here to decide which jet was lepton/photon matched
+            bool isLepPhotonMatch = false;
+            if(isLepPhotonMatch) ak8Filter[i] = false;
+        }
+
+        //Corrrolate AK8 jets and their subjets
+        unsigned int nFatJets = FatJet_LV.size();
+        unsigned int nSubJets = SubJet_LV.size();
+        std::vector<std::vector<TLorentzVector>> subjets(nFatJets);
+        for(int i = 0; i < nFatJets; ++i)
+        {
+            if(FatJet_subJetIdx1[i] >= 0 && FatJet_subJetIdx1[i] < nSubJets) subjets[i].push_back(SubJet_LV[i]);
+            if(FatJet_subJetIdx2[i] >= 0 && FatJet_subJetIdx2[i] < nSubJets) subjets[i].push_back(SubJet_LV[i]);
+        }
+
+        //Create top tagger input helpers
+        ttUtility::ConstAK4Inputs<float> ak4Inputs(Jet_LV, Jet_csvv2);
+        ak4Inputs.setFilterVector(ak4Filter);
+        ttUtility::ConstAK8Inputs<float> ak8Inputs(FatJet_LV, FatJet_deepAK8_t, FatJet_deepAK8_w, FatJet_msoftdrop, subjets);
+        ak8Inputs.setFilterVector(ak8Filter);
+        ttUtility::ConstResolvedCandInputs<float> resInputs(ResTopCand_LV, ResTopCand_discriminator, ResTopCand_j1Idx, ResTopCand_j2Idx, ResTopCand_j3Idx);
+
+        //make top tagger constituents, order matters here, ak4Inputs must not come after resInputs as the ak4Inputs are needed to build the resolved top candidates 
+        std::vector<Constituent> constituents = packageConstituents(ak4Inputs, resInputs, ak8Inputs);
+        
+        // TopTagger
+        //std::cout << "Create TopTagger object" << std::endl;
+        TopTagger tt("TopTagger.cfg", ".");
```

Comment:
you are creating a new TopTagger every loop, this is very inefficient.

See how it is done here
https://github.com/StealthStop/Framework/blob/master/Framework/include/RunTopTagger.h